### PR TITLE
refactor: simplify move legality check

### PR DESCRIPTION
**Pull request by gahtan-syarif**
reduce calls to setEngineIllegalMoveStatus function

### CODE REVIEW

File: app/src/matchmaking/match/match.cpp [hunk 1]
```diff
@@ -248,13 +248,9 @@ bool Match::playMove(Player& us, Player& them) {
...
 
-    if (best_move && !isUciMove(best_move.value())) {
-        setEngineIllegalMoveStatus(us, them, best_move, true);
-        return false;
-    }
-
-    const auto move  = best_move ? uci::uciToMove(board_, *best_move) : Move::NO_MOVE;
-    const auto legal = isLegal(move);
+    const auto move   = best_move ? uci::uciToMove(board_, *best_move) : Move::NO_MOVE;
+    const auto is_uci = isUciMove(best_move.value());
```
Comment (Disservin) [inline app/src/matchmaking/match/match.cpp:252]: This will throw an exception if no bestmove is found just use value_or("")

**Review by Disservin — CHANGES_REQUESTED** [review]

File: app/src/matchmaking/match/match.cpp (hunk 1 above)
Comment (gahtan-syarif) [reply app/src/matchmaking/match/match.cpp]: alright, fixed 👍

**Comment by Disservin** [comment]
Or mh.. then the warning is wrong..

**Comment by gahtan-syarif** [comment]
which warning?

**Comment by gahtan-syarif** [comment]
if ur talking about this i think it can just be removed https://github.com/Disservin/fastchess/blob/865610cbfc2965a5711bbe85a4ffa715d597d877/app/src/matchmaking/match/match.cpp#L410
i dont see the point of the warning since the move is adjudicated as illegal anyway, im sure the user can figure out why the move is illegal

**Comment by Disservin** [comment]
mh id rather prefer that since there's no explicit definition of move format really

**Pull request closed without merging**